packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -396,24 +407,54 @@ const SynapseGrid = forwardRef<
                 value={rowValues}
                 columns={colValues}
                 rowKey="_rowId"
-                rowClassName={({ rowData }: any) => {
-                  if (deletedRowIds.has(rowData._rowId)) {
-                    return 'row-deleted'
-                  }
-                  if (createdRowIds.has(rowData._rowId)) {
-                    return 'row-created'
-                  }
-                  if (updatedRowIds.has(rowData._rowId)) {
-                    return 'row-updated'
-                  } else return ''
-                }}
+                rowClassName={({ rowData, rowIndex }: any) =>
+                  classNames({
+                    'row-deleted': deletedRowIds.has(rowData._rowId),
+                    'row-created': createdRowIds.has(rowData._rowId),
+                    'row-updated': updatedRowIds.has(rowData._rowId),
+                    'row-valid': rowData.validationStatus === 'valid',
+                    'row-invalid': rowData.validationStatus === 'invalid',
+                    'row-unknown': rowData.validationStatus === 'unknown',
+                    'row-selected': selectedRowIndex === rowIndex,
+                  })
+                }
                 createRow={addRowToModel}
                 duplicateRow={({ rowData }: any) => ({
                   ...rowData,
                   _rowId: genId(),
                 })}
                 onChange={handleChange}
+                onActiveCellChange={({ cell }) => {
+                  setSelectedRowIndex(cell ? cell.row : null)
+                }}
               />
+              {/* Show validation messages for selected row */}
+              {selectedRowIndex !== null &&
+                rowValues[selectedRowIndex] &&
+                Array.isArray(rowValues[selectedRowIndex].validationMessages) &&
+                rowValues[selectedRowIndex].validationMessages.length > 0 && (
+                  <div
+                    style={{
+                      marginTop: 12,
+                      background: '#fffbe6',
+                      border: '1px solid #ffe58f',
+                      borderRadius: 4,
+                      padding: 10,
+                      fontSize: 13,
+                    }}
+                  >
+                    <strong>Validation Messages:</strong>
+                    {rowValues[selectedRowIndex].validationMessages.map(
+                      (msg: string, i: number) => (
+                        <FullWidthAlert
+                          variant={'warning'}
+                          isGlobal={false}
+                          description={msg}
+                        />
+                      ),
+                    )}
+                  </div>
+                )}
```

Comment:
Use the __validationResults object. Also, I was thinking just one `FullWidthAlert`.
```suggestion
              {selectedRowIndex !== null &&
                rowValues[selectedRowIndex] &&
                Array.isArray(rowValues[selectedRowIndex].__validationResults?.allValidationMessages) &&
                rowValues[selectedRowIndex].__validationResults?.allValidationMessages.length > 0 && (
                  <FullWidthAlert
                    variant='warning'
                    title='Validation Messages:'
                    description={(
                      <ul>
                        {rowValues[selectedRowIndex].validationMessages.map(
                          (msg: string) => (
                            <li key={msg}>{msg}</li>
                          ),
                        )}
                      </ul>
                    )}
                    sx={{
                      marginTop: '12px',
                    }}
                  />
                )}
```